pkg/controller/volume/persistentvolume/controller_base.go

```diff
@@ -509,6 +509,29 @@ func (ctrl *PersistentVolumeController) upgradeVolumeFrom1_2(volume *api.Persist
 	return true
 }
 
+// findDeleterParameters finds StorageClass.Parameters of storage class where
+// given volume belongs
+func (ctrl *PersistentVolumeController) findDeleterParameters(volume *api.PersistentVolume) (map[string]string, error) {
+	className := getVolumeClass(volume)
+	if className == "" {
+		return nil, nil
+	}
+
+	obj, found, err := ctrl.classes.GetByKey(className)
```

Comment:
> what will happen if StorageClass is deleted before volume is deleted

Deleter will fail and the volume enters phase Failed.

The other option would be to put StorageClass.Parameters into annotation(s) of provisioned PVs. This way, if someone reconfigures the storage appliance, e.g. changes its address, it won't be possible to delete the volume as the old address would be used. I am afraid it's suboptimal either way.